### DISCUSSION TITLE
Plugin obs-websocket 5

### DISCUSSION THREAD

**Discussion by e-Feitosa** [sircharlo/meeting-media-manager#457] [Q&A] [answered]
With the **OBS Studio update 28.0**, the **obs-websocket plugin** was updated to version 5 and the port is now 4455, so, who updates the OBS Studio can't use M³, i need a suggestion!

![image](https://user-images.githubusercontent.com/90984283/188021607-ed316b50-1013-4015-b4ba-5d955248dbbe.png)

@sircharlo

**Comment by sircharlo** [comment, marked as answer]
Until an update comes out to enable compatibility with v5, you'll need to make sure that you install the latest version of v4 (currently 4.9.1-compat) of obs-websocket to avoid any issues. You can get it from here:

https://github.com/obsproject/obs-websocket/releases/tag/5.0.1

Configure the port number and password for obs-websocket as needed.

**Reply by e-Feitosa** [reply]
> Até que uma atualização saia para permitir a compatibilidade com o v5, você precisará ter certeza de que você instalou a versão mais recente do v4 (atualmente 4.9.1-compat) de obs-websocket para evitar quaisquer problemas. Você pode obtê-lo a partir daqui:
> 
> https://github.com/obsproject/obs-websocket/releases/tag/5.0.1
> 
> Configure o número de porta e senha para obs-websocket conforme necessário.


Thank you very much for your answer. However, version 4.9.1 of the OBS-WEBSOCKT plugin does not work in OBS Studio 28.0, the new OBS Studio already comes with version 5 of the plugin as native, and when installing version 4.9.1, it will be incompatible with version 28 of the OBS, in short: until the new version of M³ comes out, it will not be possible to update OBS Studio to version 28.0.

**Reply by e-Feitosa** [reply]
Link
https://github.com/obsproject/obs-websocket/wiki/Notable-changes-between-4.x-and-5.x

**Reply by sircharlo** [reply]
> Link
> 
> https://github.com/obsproject/obs-websocket/wiki/Notable-changes-between-4.x-and-5.x

https://github.com/obsproject/obs-websocket/releases/tag/4.9.1-compat

**Reply by e-Feitosa** [reply]
> > Link
> > https://github.com/obsproject/obs-websocket/wiki/Notable-changes-between-4.x-and-5.x
> 
> https://github.com/obsproject/obs-websocket/releases/tag/4.9.1-compat

I downloaded it from this link, the previous link I sent, is the developer information about version 5. Even if you download version 4.9.1, it will not work in OBS Studio version 28.0.

**Reply by sircharlo** [reply]
Are you sure brother?

"Install this version only if you require legacy 4.9.1 protocol support **and are using OBS 28.0.0 or above**"

**Reply by e-Feitosa** [reply]
Now it worked, I re-did the installation, before, I disabled all the settings I had made in plugin 5 and disabled the plugin in M³. After the installation, I configured OBS Studio (plugin) and OBS Studio + M³ worked. Thank you very much for your patience and help. Link https://github.com/obsproject/obs-websocket/releases/tag/4.9.1-compat

🗣️ @sircharlo  **Do you have an estimated release date for the version of m³ that will accept the obs-websocket 5 plugin?**

**Reply by mtdvlpr** [reply]
I am actually in the process of a big refactor of the M³ codebase (in collaboration with @sircharlo). It is near completion and only requires further testing. The refactor includes obs-websocket v5 support and an alpha pre-release is available at [my forked repo of M³](https://github.com/mtdvlpr/meeting-media-manager/). Please note that this is a pre-release and therefore I can't guarantee that everything works as intended. You can test it out on your own device (not the Kingdom Hall computer) to see if it's any good. Any feedback would be much appreciated!

**Reply by e-Feitosa** [reply]
> I am actually in the process of a big refactor of the M³ codebase (in collaboration with @sircharlo). It is near completion and only requires further testing. The refactor includes obs-websocket v5 support and an alpha pre-release is available at [my forked repo of M³](https://github.com/mtdvlpr/meeting-media-manager/). Please note that this is a pre-release and therefore I can't guarantee that everything works as intended. You can test it out on your own device (not the Kingdom Hall computer) to see if it's any good. Any feedback would be much appreciated!

I've made some tests with the plugin v5. Result, if the OBS Studio is open, the M³ doesn't open, it stays stopped on the screen looking for an update. With OBS Studio closed, M3 opens, then I start OBS Studio, and it connects to OBS Studio, but, the SCENES don't show up in M3, according to the print with the plugin 4.9

![image](https://user-images.githubusercontent.com/90984283/189721007-028113f2-9cf2-46ea-a1c2-679c98865c42.png)


![image](https://user-images.githubusercontent.com/90984283/189720492-ea9d4cbb-2f19-4567-94b9-2cc85836e44a.png)

**Reply by mtdvlpr** [reply]
You're testing the obs-websocket v5 with the current M³. You should download the alpha M³ version from [my forked repo](https://github.com/mtdvlpr/meeting-media-manager/releases/latest). That one has v5 support.

**Reply by mtdvlpr** [reply]
But don't download it on the kingdom hall computer, since it is not a stable version. It's purely for testing.

**Reply by e-Feitosa** [reply]
> You're testing the obs-websocket v5 with the current M³. You should download the alpha M³ version from [my forked repo](https://github.com/mtdvlpr/meeting-media-manager/releases/latest). That one has v5 support.

I was unable to access this page with the previous link. Now I managed to access to download the program and am testing, see a problem here.
¹ When choosing the option to download the media, I am getting this error. 
² With regard to the plugin, it worked perfectly. And the program is very fast with respect to any changes in the settings.

![image](https://user-images.githubusercontent.com/90984283/189748728-b51d9fe8-7de5-4252-b8cf-6e536511e5a9.png)

### Describe the bug
A clear and concise description of what the bug is.

### To Reproduce
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Do '....'
4. See error

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots
If possible, add screenshots to help explain your problem.

### System specs
- Windows_NT 10.0.22000
- M³ v22.9.0-alpha.4

### Additional context
Add any other context about the problem here.

### Anonymized `prefs.json`
```
{
  "app": {
    "autoOpenFolderWhenDone": false,
    "autoQuitWhenDone": false,
    "autoRunAtBoot": false,
    "autoStartSync": false,
    "congregationName": "Testing",
    "disableHardwareAcceleration": false,
    "localAppLang": "pt",
    "localOutputPath": "***",
    "obs": {
      "enable": true,
      "port": "***",
      "password": "***",
      "mediaScene": "Midias",
      "cameraScene": "P a l c o",
      "useV4": false
    },
    "outputFolderDateFormat": "YYYY-MM-DD",
    "theme": "system"
  },
  "cong": {
    "server": null,
    "port": null,
    "user": null,
    "password": null,
    "dir": null
  },
  "media": {
    "enableMediaDisplayButton": true,
    "enableMp4Conversion": false,
    "enablePp": false,
    "enableVlcPlaylistCreation": false,
    "excludeLffi": false,
    "excludeLffiImages": false,
    "excludeTh": true,
    "hideMediaLogo": false,
    "keepOriginalsAfterConversion": false,
    "lang": "T",
    "langUpdatedLast": "2022-09-12T19:39:19.635Z",
    "maxRes": "720p",
    "ppBackward": null,
    "ppForward": null,
    "preferredOutput": "window"
  },
  "meeting": {
    "enableMusicButton": false,
    "enableMusicFadeOut": false,
    "musicFadeOutTime": 5,
    "musicFadeOutType": "smart",
    "musicVolume": 100,
    "mwDay": 2,
    "mwStartTime": "19:30",
    "weDay": 5,
    "weStartTime": "19:00"
  },
  "__internal__": {
    "migrations": {
      "version": "22.9.0-alpha.4"
    }
  }
}
```

### Full error log
```
{
  "1663013397179": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663013397189": [
    "No MW media data found for 2022-09-14!"
  ],
  "1663013397374": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663013397512": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'",
    "No WE meeting data found for 2022-09-17!"
  ],
  "1663013692458": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'",
    "No MW media data found for 2022-09-14!"
  ],
  "1663013692610": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663013692738": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'",
    "No WE meeting data found for 2022-09-17!"
  ]
}
```
### Error details
```
{
  "stack": "Error: No WE meeting data found for 2022-09-17!\n    at app://./_nuxt/5ebeb5a.js:1:143878\n    at v (app://./_nuxt/f0cc2a3.js:2:84198)\n    at Generator._invoke (app://./_nuxt/f0cc2a3.js:2:83986)\n    at Generator.next (app://./_nuxt/f0cc2a3.js:2:84627)\n    at n (app://./_nuxt/be57a52.js:2:8117)\n    at Q (app://./_nuxt/be57a52.js:2:8320)",
  "message": "No WE meeting data found for 2022-09-17!",
  "__sentry_captured__": true
}
```

**Reply by mtdvlpr** [reply]
I think I know what caused it. I tried to optimize this bit of code, but it seems that it did not work for Windows at least. Will release a (potential) fix momentarily!

**Reply by mtdvlpr** [reply]
Released a new version. It should automatically update yours.

**Reply by mtdvlpr** [reply]
Please let me know if this solves your problem

**Reply by e-Feitosa** [reply]
> Please let me know if this solves your problem

### Describe the bug
A clear and concise description of what the bug is.

### To Reproduce
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Do '....'
4. See error

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots
If possible, add screenshots to help explain your problem.

![image](https://user-images.githubusercontent.com/90984283/189757514-57132a93-ad86-4ff7-98d3-054125053c38.png)

### System specs
- Windows_NT 10.0.22000
- M³ v22.9.0-alpha.5

### Additional context
Add any other context about the problem here.

### Anonymized `prefs.json`
```
{
  "app": {
    "autoOpenFolderWhenDone": false,
    "autoQuitWhenDone": false,
    "autoRunAtBoot": true,
    "autoStartSync": true,
    "congregationName": "Testing",
    "disableHardwareAcceleration": false,
    "localAppLang": "en",
    "localOutputPath": "***",
    "obs": {
      "enable": true,
      "port": "***",
      "password": "***",
      "mediaScene": "Midias",
      "cameraScene": "P a l c o",
      "useV4": false
    },
    "outputFolderDateFormat": "YYYY-MM-DD - dddd",
    "theme": "system"
  },
  "cong": {
    "server": null,
    "port": null,
    "user": null,
    "password": null,
    "dir": null
  },
  "media": {
    "enableMediaDisplayButton": true,
    "enableMp4Conversion": false,
    "enablePp": false,
    "enableVlcPlaylistCreation": false,
    "excludeLffi": false,
    "excludeLffiImages": false,
    "excludeTh": true,
    "hideMediaLogo": false,
    "keepOriginalsAfterConversion": false,
    "lang": "T",
    "langUpdatedLast": "2022-09-12T20:22:29.548Z",
    "maxRes": "720p",
    "ppBackward": null,
    "ppForward": null,
    "preferredOutput": "window"
  },
  "meeting": {
    "enableMusicButton": true,
    "enableMusicFadeOut": true,
    "musicFadeOutTime": 60,
    "musicFadeOutType": "smart",
    "musicVolume": 29,
    "mwDay": 2,
    "mwStartTime": "19:30",
    "weDay": 5,
    "weStartTime": "19:00"
  },
  "__internal__": {
    "migrations": {
      "version": "22.9.0-alpha.5"
    }
  }
}
```

### Full error log
```
{
  "1663016254309": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663016254320": [
    "No MW media data found for 2022-09-14 - Wednesday!"
  ],
  "1663016254479": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663016254607": [
    "Error: ENOENT: no such file or directory, open '\\sql-wasm.wasm'"
  ],
  "1663016254608": [
    "No WE meeting data found for 2022-09-17 - Saturday!"
  ]
}
```
### Error details
```
{
  "stack": "Error: No WE meeting data found for 2022-09-17 - Saturday!\n    at app://./_nuxt/35654db.js:1:143929\n    at v (app://./_nuxt/9af5c8b.js:2:84198)\n    at Generator._invoke (app://./_nuxt/9af5c8b.js:2:83986)\n    at Generator.next (app://./_nuxt/9af5c8b.js:2:84627)\n    at n (app://./_nuxt/462a916.js:2:8117)\n    at Q (app://./_nuxt/462a916.js:2:8320)",
  "message": "No WE meeting data found for 2022-09-17 - Saturday!",
  "__sentry_captured__": true
}
```

**Reply by mtdvlpr** [reply]
Okay, the issue is deeper than I thought. I'll have to look into it some other time. I will keep you posted on the progress here!

**Reply by mtdvlpr** [reply]
@e-Feitosa , I have fixed the problem in alpha.6 :)

**Reply by e-Feitosa** [reply]
> @e-Feitosa , corrigi o problema em alfa.6 :)

Perfect, now it's working. Another problem I noticed, is related to the added image, if the image has a special special character or symbol, it does not play on the second screen as shown in the example, the second screen continues with the year text. 

![image](https://user-images.githubusercontent.com/90984283/189922034-3ba604ba-133f-4f12-b1c8-49e5095652bc.png)

**Reply by mtdvlpr** [reply]
Fixed in latest release!

**Reply by e-Feitosa** [reply]
OK, i updated and it still continues without displaying the image when it has a special character, in the example, the image has (1), and this is preventing it from being displayed.

![image](https://user-images.githubusercontent.com/90984283/190010214-30a8ea5d-61da-4cb3-987b-0292c9bcf0b6.png)

**Reply by mtdvlpr** [reply]
My bad. It's fixed in alpha.8 (this time I checked)

**Reply by mtdvlpr** [reply]
You might have to remove the images and add them again to the media list

**Reply by mtdvlpr** [reply]
So they get parsed again

**Reply by e-Feitosa** [reply]
Thanks again, it worked, just one detail, if I copy an image and paste it inside the folder and the name of the file is the same as the examples above with "( )", the program does not play. When I add the image by the program, it removes the "( )" and the image works perfectly.

**Reply by mtdvlpr** [reply]
I will add the parsing to manually added files in next release!

**Comment by Chipperchap** [comment]
Websocket is working but M3 gives an error that it is not working. Please see video [HERE](https://youtu.be/Olku4WUB7Cc)

**Reply by mtdvlpr** [reply]
Hmm, I will look into this false positive for sure! I won't rush with a release for it though, since it is a false positive.

**Reply by mtdvlpr** [reply]
Hi @Chipperchap, I could not pinpoint the exact issue you were experiencing, but I have improved the overall logging functionality for OBS, so it might be resolved in the latest version.

**Comment by Chipperchap** [comment]
Cannot delete media (unwanted for the week due to replaced with GB Update # 6) (...want it hidden) 
...and will not display custom media (GB # 6).  
Please see video [HERE](https://youtu.be/ToqJJ3XBgNQ).

**Reply by mtdvlpr** [reply]
Fix has been released! Make sure to remove the broken file (the dot (,) in .mp4 had been removed)

**Reply by e-Feitosa** [reply]
Working perfectly with OBS Studio v28 and Plugin v5

**Reply by e-Feitosa** [reply]
Just commenting on a problem that has now arisen that didn't happen before. If you open the M3 program first, the error occurs with OBS Studio and to work, you have to go in the settings to disable and enable compatibility with OBS Studio. If you open OBS Studio and then the M3 program, it works normally. When is this a problem? When we leave the programs to start along with Windows, otherwise, just open OBS Studio first and then M3.

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots
If possible, add screenshots to help explain your problem.

### System specs
- Windows_NT 10.0.22000
- M³ v22.9.0-alpha.9

### Additional context
Add any other context about the problem here.

### Anonymized `prefs.json`
```
{
  "app": {
    "autoOpenFolderWhenDone": false,
    "autoQuitWhenDone": false,
    "autoRunAtBoot": true,
    "autoStartSync": true,
    "congregationName": "Testing",
    "disableHardwareAcceleration": false,
    "localAppLang": "pt",
    "localOutputPath": "***",
    "obs": {
      "enable": true,
      "port": "***",
      "password": "***",
      "mediaScene": "Midias",
      "cameraScene": "P a l c o",
      "useV4": false
    },
    "outputFolderDateFormat": "YYYY-MM-DD - dddd",
    "theme": "system"
  },
  "cong": {
    "server": null,
    "port": null,
    "user": null,
    "password": null,
    "dir": null
  },
  "media": {
    "enableMediaDisplayButton": true,
    "enableMp4Conversion": false,
    "enablePp": false,
    "enableVlcPlaylistCreation": false,
    "excludeLffi": false,
    "excludeLffiImages": false,
    "excludeTh": true,
    "hideMediaLogo": false,
    "keepOriginalsAfterConversion": false,
    "lang": "T",
    "langUpdatedLast": "2022-09-14T13:51:02.750Z",
    "maxRes": "720p",
    "ppBackward": null,
    "ppForward": null,
    "preferredOutput": "window"
  },
  "meeting": {
    "enableMusicButton": true,
    "enableMusicFadeOut": true,
    "musicFadeOutTime": 60,
    "musicFadeOutType": "smart",
    "musicVolume": 10,
    "mwDay": 2,
    "mwStartTime": "19:30",
    "weDay": 5,
    "weStartTime": "19:00"
  },
  "__internal__": {
    "migrations": {
      "version": "22.9.0-alpha.9"
    }
  }
}
```

### Full error log
```
{
  "1663167834476": [
    ""
  ],
  "1663167964751": [
    ""
  ],
  "1663167964883": [
    ""
  ]
}
```
### Error details
```
{
  "stack": "Error\n    at B.onError (app://./_nuxt/60d2110.js:5:396388)",
  "code": -1,
  "__sentry_captured__": true
}
```

**Reply by mtdvlpr** [reply]
Hi @e-Feitosa , this has always been the case to my knowledge. It might be that the initial load now takes a little longer and therefore M3 starts up later than OBS.

**Comment by e-Feitosa** [comment]
This error below is occurring after updating OBS Studio to version V.28.0.2 (version ... 0.2). When you open M3 it connects normally, only when you access the M3 settings, an error screen appears, but even so, the program continues to function normally with OBS Studio v.28 and plugin v.5.

![image](https://user-images.githubusercontent.com/90984283/192041099-b17cc3e4-6223-4ad3-80c8-c25bf0c39902.png)


### System specs
- Windows_NT 10.0.22000
- M³ v22.9.0-alpha.12

### Additional context
Add any other context about the problem here.

### Anonymized `prefs.json`
```
{
  "app": {
    "autoOpenFolderWhenDone": false,
    "autoQuitWhenDone": false,
    "autoRunAtBoot": false,
    "autoStartSync": false,
    "congregationName": "Testing",
    "disableHardwareAcceleration": false,
    "localAppLang": "pt",
    "localOutputPath": "***",
    "obs": {
      "enable": true,
      "port": "***",
      "password": "***",
      "mediaScene": "Midias",
      "cameraScene": "P a l c o",
      "useV4": false
    },
    "outputFolderDateFormat": "YYYY-MM-DD - dddd",
    "theme": "dark"
  },
  "cong": {
    "server": "",
    "port": "***",
    "user": null,
    "password": null,
    "dir": "***"
  },
  "media": {
    "enableMediaDisplayButton": true,
    "enableMp4Conversion": false,
    "enablePp": false,
    "enableVlcPlaylistCreation": false,
    "excludeLffi": false,
    "excludeLffiImages": false,
    "excludeTh": true,
    "hideMediaLogo": false,
    "keepOriginalsAfterConversion": false,
    "lang": "T",
    "langUpdatedLast": "2022-09-19T17:44:07.942Z",
    "maxRes": "720p",
    "ppBackward": null,
    "ppForward": null,
    "preferredOutput": "window"
  },
  "meeting": {
    "enableMusicButton": true,
    "enableMusicFadeOut": true,
    "musicFadeOutTime": 60,
    "musicFadeOutType": "smart",
    "musicVolume": 100,
    "mwDay": 2,
    "mwStartTime": "19:30",
    "weDay": 5,
    "weStartTime": "19:00"
  },
  "__internal__": {
    "migrations": {
      "version": "22.9.0-alpha.12"
    }
  }
}
```

### Full error log
```
{
  "1663960057919": [
    "Socket not identified"
  ]
}
```
### Error details
```
{
  "stack": "Error: Socket not identified\n    at B.message (app://./_nuxt/59ca980.js:5:399864)\n    at B.call (app://./_nuxt/59ca980.js:5:398039)\n    at app://./_nuxt/6ad0df2.js:1:158734\n    at v (app://./_nuxt/0460e21.js:2:83451)\n    at Generator._invoke (app://./_nuxt/0460e21.js:2:83239)\n    at Generator.next (app://./_nuxt/0460e21.js:2:83880)\n    at n (app://./_nuxt/59ca980.js:2:2329)\n    at Q (app://./_nuxt/59ca980.js:2:2532)",
  "message": "Socket not identified",
  "__sentry_captured__": true
}
```

**Reply by mtdvlpr** [reply]
I have a fix for this false positive. Will be included in next release!

**Comment by e-Feitosa** [comment]
@sircharlo Recently friends have reported about the difficulty they are having when OBS Studio updates, many don't know how to solve it by installing the v.4x plugin and then disable the v.5 plugin and configure again the v.4x plugin. Some have managed to solve it after explaining this option, however, many are not succeeding. This incompatibility is creating a problem for those who do not master these issues. Would it be possible to put a note in the current version informing of this option to disable the plung v.5 and enable the plugin v.4x? It would be something temporary until the testing of the new version is finished.
